api/types/types.go

```diff
@@ -145,10 +146,10 @@ type PacketInjection struct {
 	BasicResource    `yaml:",inline"`
 	Src              string `yaml:"Src"`
 	Dst              string `yaml:"Dst"`
-	SrcIP            string `yaml:"SrcIP"`
-	DstIP            string `yaml:"DstIP"`
-	SrcMAC           string `yaml:"SrcMAC"`
-	DstMAC           string `yaml:"DstMAC"`
+	SrcIP            string `valid:"isCIDR" yaml:"SrcIP"`
+	DstIP            string `valid:"isCIDR" yaml:"DstIP"`
+	SrcMAC           string `valid:"isMAC" yaml:"SrcMAC"`
+	DstMAC           string `valid:"isMAC" yaml:"DstMAC"`
 	SrcPort          int64  `yaml:"SrcPort"`
```

Comment:
You may also consider validating port range - you can take some ideas from: https://golang.org/src/net/port.go